src/main/java/com/sda/dao/implementation/CategoryDaoImplementation.java

```diff
@@ -0,0 +1,38 @@
+package com.sda.dao.implementation;
+
+import com.sda.dao.CategoryDao;
+import com.sda.entity.Category;
+import org.hibernate.Session;
+
+import java.io.Serializable;
+
+public class CategoryDaoImplementation implements CategoryDao {
+
+    private  Session session;
```

Comment:
For connecting to a database you should create session, then create transaction **only then save object** and commit transaction after that.